aiida_quantumespresso/workflows/pw/base.py

```diff
@@ -100,7 +99,8 @@ def define(cls, spec):
         spec.exit_code(203, 'ERROR_INVALID_INPUT_RESOURCES',
             message='Neither the `options` nor `automatic_parallelization` input was specified.')
         spec.exit_code(204, 'ERROR_INVALID_INPUT_RESOURCES_UNDERSPECIFIED',
-            message='The `metadata.options` did not specify both `resources.num_machines` and `max_wallclock_seconds`.')
+            message='The `metadata.options` did not specify both `resources.num_machines` and `max_wallclock_seconds`. '
+                    'This exit status has been deprecated as the check it corresponded too was incorrect.')
```

Comment:
Also, shouldn't this exit code also be deprecated for the `PhBaseWorkChain`?

Reply:
Honestly, I don't know. This is unknown terrain as for us it hasn't come up before. I did think about this way back when introducing the concept of exit codes and decided to store the actual exit message on the node. So each node has the `exit_message` attribute which contains the message of the exit code as it was defined at runtime, in addition to the `exit_status` which stores the corresponding integer. So even if we change the implementation, existing nodes will still reflect the original message through things like `verdi process status/list` etc. Still, the integer remains the same, so I am not sure if we should repurpose it, because if you then perform a query on it, you might get processes that failed for a different reason but it seems to be the same one (based off of just the status and ignoring the message).

So in short, I am not sure. I don't think we should repurpose the exit code, so better to leave it in because if we remove it, at some point we forget it existed and might use the integer again. Then again, without a deprecation pathway they might start to accumulate over time and get in the way. Open to ideas and suggestions

Reply:
Hmm, good points, thanks for the explanation. Just off the top of my head I would consider adding a `deprecated_exit_status` class variable to the process classes where you can keep track of deprecated exit codes, so the class can raise in case a new exit code is defined on the spec with an already deprecated `exit_status`. This would definitely be shorter than keeping them all in the `define` method.

Maybe it would be best to open a discussion on the `aiida-core` discussions board?

Reply:
> Maybe it would be best to open a discussion on the aiida-core discussions board?

yes, let's

Reply:
https://github.com/aiidateam/aiida-core/discussions/4933